server/kv/etcd_kv.go

```diff
@@ -93,11 +89,12 @@ func (kv *etcdKVBase) Save(key, value string) error {
 	txn := NewSlowLogTxn(kv.client)
 	resp, err := txn.Then(clientv3.OpPut(key, value)).Commit()
 	if err != nil {
-		log.Error("save to etcd meet error", zap.String("key", key), zap.String("value", value), errs.ZapError(errs.ErrEtcdKVSave, err))
-		return errors.WithStack(err)
+		e := errs.ErrEtcdKVPut.Wrap(err).GenWithStackByCause()
+		log.Error("save to etcd meet error", zap.String("key", key), zap.String("value", value), errs.ZapError(e))
+		return e
 	}
 	if !resp.Succeeded {
-		return errors.WithStack(errTxnFailed)
+		return errs.ErrEtcdTxn.FastGenByArgs()
```

Comment:
This error has a `%s` placeholder. is it ok to pass no argument?

Reply:
updated